canister/src/tests.rs

```diff
@@ -716,3 +716,35 @@ async fn test_syncing_with_next_block_headers() {
 
     assert!(catch_unwind(verify_synced).is_err());
 }
+
+#[async_std::test]
+async fn test_cycles_burnt() {
```

Comment:
This test is testing that we are incrementing the metrics, but it's not actually testing that we're burning cycles. I think we'd need an e2e test for that, otherwise we'll never exercise the code path where real cycles are actually burnt.

Reply:
The test is added now, this was the best I could do it in order not to make the test flaky. It confirms that cycles are burnt, after initializing canister with a lot of cycles.

Reply:
PTAL